.travis.yml

```diff
@@ -28,17 +28,17 @@ addons:
       - llvm-toolchain-precise
 
 env:
-  - COMPILER=clang++-3.6 CMAKE_GENERATOR="Ninja"
-  - COMPILER=clang++-3.6 CMAKE_GENERATOR="Unix Makefiles"
-
-  - COMPILER=clang++-3.7 CMAKE_GENERATOR="Unix Makefiles"
-  - COMPILER=clang++-3.7 CMAKE_GENERATOR="Ninja"
-
-  - COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
-  - COMPILER=clang++-3.8 CMAKE_GENERATOR="Ninja"
-
-  - COMPILER=gcc CMAKE_GENERATOR="Ninja"
-  - COMPILER=gcc CMAKE_GENERATOR="Unix Makefiles"
+  # Test all generators and Clang + GCC
+  - TEST=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
+  - TEST=true COMPILER=gcc         CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=gcc         CMAKE_GENERATOR="Unix Makefiles"
+
+  # Benchmarks
+  - BENCHMARK=true COMPILER=clang++-3.6 CMAKE_GENERATOR="Unix Makefiles"
+  - BENCHMARK=true COMPILER=clang++-3.7 CMAKE_GENERATOR="Unix Makefiles"
+  - BENCHMARK=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
+  - BENCHMARK=true COMPILER=gcc         CMAKE_GENERATOR="Unix Makefiles"
```

Comment:
> I don't think that's an issue, we just have to push them to separate directories. For example, clang-3.5/ gets all the benchmarks that were run on Clang 3.5, etc..

I was more concerned about the fact jobs run in parallel, so for example it might happen that jobs A and B clone gh-pages and than both commit their changes. Then A pushes before B and so, when B tries to push, git would complain. There might be a way to tell git how to deal with it, but I wouldn't know how.

More importantly, if several jobs are pushing data to gh-pages, which job will build the front page that shows everything?

> I certainly think this is out of scope for Metabench

I agree.

> it's useful to know how a metaprogramming library performs on a specific compiler

It definitely is, but it also adds a lot of complexity. All I'm saying is we got to start thinking about it now before we bite more than we can chew.